src/components/Shared/Filters/FilterListItem.tsx

```diff
@@ -1,57 +1,75 @@
 import { ListItem, ListItemText } from '@material-ui/core';
 import React from 'react';
-import { Filter } from './Filter';
+import { ContactFilterValue } from '../../Contacts/ContactFilters/ContactFilters';
+import {
+  Filter,
+  CheckboxFilter,
+  DaterangeFilter,
+  MultiselectFilter,
+  NumericRangeFilter,
+  RadioFilter,
+  TextFilter,
+  DateRangeInput,
+  NumericRangeInput,
+} from '../../../../graphql/types.generated';
 import { FilterListItemCheckbox } from './FilterListItemCheckbox';
 import { FilterListItemDateRange } from './FilterListItemDateRange';
 import { FilterListItemMultiselect } from './FilterListItemMultiselect';
+import { FilterListItemNumericRange } from './FilterListItemNumericRange';
 import { FilterListItemSelect } from './FilterListItemSelect';
 import { FilterListItemTextField } from './FilterListItemTextField';
 
 interface Props {
   filter: Filter;
-  value?: boolean | string | Array<string>;
-  onUpdate: (value?: boolean | string | Array<string>) => void;
+  value?: ContactFilterValue;
+  onUpdate: (value?: boolean | string | Array<string> | DateRangeInput) => void;
 }
 
 export const FilterListItem: React.FC<Props> = ({
   filter,
   value,
   onUpdate,
 }: Props) => {
-  return filter.type === 'text' ? (
+  return (filter as TextFilter).__typename === 'TextFilter' ? (
```

Comment:
There should be a way to make TS happy here without casting. Worst case we could use this https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates. But there should be a better way. What does it do without the casts? My hunch is it might just work but there's probably something wrong with it if you had to cast things.

Reply:
I think it doesn't work because of the way `ContactFilters.graphql` is set up... `__typename` is not part of the main filter type, but it gets added among the type-specific properties.  I can try reworking this and see if there is a better solution.